hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java

```diff
@@ -290,7 +290,7 @@ public void init() {
       //replace environment variables to support multi-node execution
       prefix = resolvePrefix(prefix);
     }
-    if (duration != null) {
+    if (duration != null && allowDuration()) {
```

Comment:
Let's reset `duration = null` -- this way we can omit `allowDuration()` call in other places.

```java
if (!allowDuration()) {
  LOG.warn("--duration is ignored");
  duration = null;
}
```

Reply:
Thanks @adoroszlai .
I have updated it.
Test:
`
./bin/ozone freon ombg -v test8 -n 25 --duration 1m
`
Results:
```
2024-09-30 22:24:44,191 [main] INFO impl.MetricsConfig: Loaded properties from hadoop-metrics2.properties
2024-09-30 22:24:44,265 [main] INFO impl.MetricsSystemImpl: Scheduled Metric snapshot period at 10 second(s).
2024-09-30 22:24:44,265 [main] INFO impl.MetricsSystemImpl: ozone-freon metrics system started
2024-09-30 22:24:44,326 [main] WARN freon.BaseFreonGenerator: --duration is ignored
2024-09-30 22:24:44,326 [main] INFO freon.BaseFreonGenerator: Executing test with prefix wargf41605 and number-of-tests 25

 100.00% |█████████████████████████████████████████████████████████████████████████████████████████████████████|  25/25 Time: 0:00:02|  
9/30/24 10:24:46 PM ============================================================

-- Timers ----------------------------------------------------------------------
bucket-create
             count = 25
         mean rate = 24.51 calls/second
     1-minute rate = 0.00 calls/second
     5-minute rate = 0.00 calls/second
    15-minute rate = 0.00 calls/second
               min = 3.19 milliseconds
               max = 33.62 milliseconds
              mean = 18.74 milliseconds
            stddev = 11.70 milliseconds
            median = 14.02 milliseconds
              75% <= 33.15 milliseconds
              95% <= 33.58 milliseconds
              98% <= 33.62 milliseconds
              99% <= 33.62 milliseconds
            99.9% <= 33.62 milliseconds


Total execution time (sec): 2
Failures: 0
Successful executions: 25
```